[WIP] Docker Compose Overrides for MonoRepo

anabeto93 opened the pull request:
### Summary

<!-- ✍️ A clear and concise description...-->

#### Related Issues / PR's

<!-- ✍️ Add the issues being resolved here and related PR's where applicable  -->

#### Screenshots

NA

<!-- ✍️ Add screenshots of before and after changes where applicable-->

#### Affected Areas and Manually Tested Areas

<!-- ✍️ Add details of blast radius and dev testing areas where applicable-->

anabeto93 committed:
override to work with current monorepo.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/SigNoz/signoz?pullRequest=4960) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/SigNoz/signoz?pullRequest=4960) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/SigNoz/signoz?pullRequest=4960) it.</sub>

anabeto93 committed:
limit logging.

YounixM committed:
feat: onboarding v2 base setup

YounixM committed:
feat: add questionaire components (#5998)

* feat: add questionaire components

* feat: update css

* feat: delete icon svgs and update css

* feat: update component names

YounixM committed:
feat: maintain state and add log events

anabeto93 committed:
changed the outdated docker-compose.yml config versions.

anabeto93 committed:
added swarm.

anabeto93 committed:
linting, beyond this, I am not sure.

anabeto93 committed:
additional linting.

ahmadshaheer committed:
fix: fix the race condition resulting in switching between views not working properly

ahmadshaheer committed:
chore: improve comments

ahmadshaheer committed:
fix: add v4 to default alert objects to fix the issue of incorrect query response due to v3 endpoint

ahmadshaheer committed:
Revert "fix: add v4 to the new alert payload (#6090)"

This reverts commit 5651d694854cd77964b9225af7aecb420c9c1fb8.

vikrantgupta25 committed:
chore: change the log background to same as severity text with some opacity (#6217)

* chore: change the log background to same as severity text with some opacity

* chore: make the log background changes for column renderer

* chore: remove the backdrop mask from logs details drawer

* chore: fix tests

rindrics committed:
chore: run all query-service tests (#6098)

vikrantgupta25 committed:
fix: issues with the logs where clause filter (#6198)

* fix: flicker in logs explorer page

* fix: added inline code comments

* fix: edit rules css should be scoped inside the container

* fix: add explicit space after adding key and operators to help users type and create the tag

* fix: do not wait for api to complete for the user to enter the query

---------

Co-authored-by: Srikanth Chekuri <srikanth.chekuri92@gmail.com>

srikanthccv committed:
chore: update query builder to use 5min/30min aggregation tables (#5679)

SagarRajput-7 committed:
fix: fixed threshold for columns with units (#6079)

* fix: fixed threshold for columns with units

* fix: added interunit and category conversion for handling threshold across different unit types

* fix: added invalid comparison text and removed unsupported unit categories

* fix: cleanup and multiple threshold and state change handling

* fix: restrict category select to only columnUnit group

* fix: restricted column name from threshold option

* fix: removed console log

* fix: resolved comments and some refactoring

* fix: resolved comments and some refactoring

SagarRajput-7 committed:
fix: removed selectedValue & uuid from exported and copied dashboard json (#6145)

* fix: removed selectedValue from exported and copied dashboard json

* fix: added uuid validation to not expect empty uuid

* fix: removed uuid from export and copied dashboard json

* fix: resolved comment and removed uuid when empty

* fix: renamed function

SagarRajput-7 committed:
chore: added doc link for kafka on getStarted for non-cloud users (#6222)

* chore: added doc link for kafka on getStarted for non-cloud users

* chore: added link with utm parameter

vikrantgupta25 committed:
chore: remove the required check from organisation name (#6225)

ahmadshaheer committed:
fix: send float number in args array in case of timeshift formula

nityanandagohain committed:
feat: move resource qb to its own package and use common options (#6238)

* feat: move resource qb to its own package and use common options

* fix: add remaining files

* fix: remove redundant struct

srikanthccv committed:
chore: split migrator job to sync and async (#6107)

saichander17 committed:
feat: enable offline functionality for frontend (#6152)

Co-authored-by: Vikrant Gupta <vikrant.thomso@gmail.com>

YounixM committed:
feat: tooltip plugin to show series data in tooltip (#6194)

* feat: tooltip plugin to show series data in tooltip

* feat: send anomaly function as last function

* feat: default z_score_threshold to 3

* fix: anomaly function not applied on initial load

* feat: maintain select alert type on reload

* feat: maintain select alert type on reload

* chore: update events to handle anomaly alert interactions

YounixM committed:
feat: integrate update profile and invite users api

YounixM committed:
feat: remove hardcode is darkmode value from get anomaly data func (#6245)

makeavish committed:
fix: name is optional in bulk invite API (#6246)

YounixM committed:
feat: handle errors for profiles and invite users api

YounixM committed:
feat: handle redirection after onboarding

YounixM committed:
feat: add org onboarding preference (#6248)

makeavish committed:
Fix/bulk invite api error response (#6247)

YounixM committed:
feat: add get and update apis for org and user preferences

dependabot[bot] committed:
chore(deps): bump micromatch from 4.0.5 to 4.0.8 in /frontend (#5817)

Bumps [micromatch](https://github.com/micromatch/micromatch) from 4.0.5 to 4.0.8.
- [Release notes](https://github.com/micromatch/micromatch/releases)
- [Changelog](https://github.com/micromatch/micromatch/blob/master/CHANGELOG.md)
- [Commits](https://github.com/micromatch/micromatch/compare/4.0.5...4.0.8)

---
updated-dependencies:
- dependency-name: micromatch
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

srikanthccv committed:
chore: add missing shiftby in alert rule (#6239)

srikanthccv committed:
chore: enable anomaly detection for ee/paid plan (#6243)

srikanthccv committed:
chore: update hosts list to use pre-aggregated data table dynamically (#6227)

srikanthccv committed:
chore: add k8s pods list (#6229)

srikanthccv committed:
chore: add k8s nodes, namespaces, and cluster list (#6230)

srikanthccv committed:
chore: fix access for downtime schedules (#6255)

YounixM committed:
feat: update the response data type

vikrantgupta25 committed:
fix: explicitly return the empty slice for variables query (#6258)

makeavish committed:
chore: remove facing issues button (#6256)

shivanshuraj1333 committed:
Fix api query context (#6268)

* chore: fix naming api query context

sharpshooter90 committed:
feat: add Request Dashboard button and improve dashboard list styles (#6251)

* feat: add Request Dashboard button and improve dashboard list styles

* feat: support for empty state and dashboard list state

therealpandey committed:
feat(integrations): whitelist /deployments/me (#6275)

vikrantgupta25 committed:
fix: arithmetic operators are removed from Dashboard query builder formulas (#6276)

* fix: the redirect from dashboard landing page to edit removing arithmetic operations

* fix: the url for the dashboard edit widget

raj-k-singh committed:
Chore: upgrade signoz otel collector dependency to v0.111.2 (#6257)

* chore: upgrade signoz-otel-collector dependencies to v0.111.2

* chore: update references to otel-collector types in collector simulator

* chore: escape '$' as '$$$' and not '$$' in generated pipeline collector config

* chore: update go.sum entry for logstransformprocessor

* chore: some more go.sum updates to get build working

---------

Co-authored-by: Srikanth Chekuri <srikanth.chekuri92@gmail.com>

raj-k-singh committed:
Fix: logs pipelines: ensure special characters in pipeline identifiers don't result in bad collector config names (#6259)

* chore: add test validating pipe char in pipeline alias doesnt break collector config

* fix: pipelines collector conf: ensure bad names are not generated

srikanthccv committed:
chore: bump signoz-otel-collector version (#6290)

YounixM committed:
feat: feedback updates

YounixM committed:
feat: feedback updates

prashant-shahi committed:
chore(signoz): pin versions: SigNoz 0.57.0

Signed-off-by: Prashant Shahi <prashant@signoz.io>

prashant-shahi committed:
chore: go mod tidy

Signed-off-by: Prashant Shahi <prashant@signoz.io>

YounixM committed:
feat: update to use v2 instance

YounixM committed:
feat: handle linear to exponential conversion for logs, services and hosts

raj-k-singh committed:
Chore: bump signoz otel collector dependency to 0.111.5 (#6302)

* chore: bump signoz-otel-collector dependency version to 0.111.5

* chore: logs filter suggestions: update import for ResourceHierarchy from signoz-otel-collector

prashant-shahi committed:
Merge branch 'develop' into release/signoz-0.57.0

Commit:
fix: fixed the step interval which was being perculated to list view (#6260)

* fix: fixed the step interval which was being perculated to list view

* fix: fixed the step interval which was being perculated to list view

* fix: fixed the step interval which was being perculated to list view

* fix: fixed the step interval which was being perculated to list view

* fix: fixed the step interval which was being perculated to list view

* fix: fixed the step interval which was being perculated to list view

* chore: bump signoz-otel-collector version (#6290)

* Chore: bump signoz otel collector dependency to 0.111.5 (#6302)

* chore: bump signoz-otel-collector dependency version to 0.111.5

* chore: logs filter suggestions: update import for ResourceHierarchy from signoz-otel-collector

* fix: fixed the step interval which was being perculated to list view

---------

Co-authored-by: Srikanth Chekuri <srikanth.chekuri92@gmail.com>
Co-authored-by: Raj Kamal Singh <1133322+raj-k-singh@users.noreply.github.com>

prashant-shahi committed:
Merge pull request #6300 from SigNoz/release/signoz-0.57.0

Release/signoz 0.57.0

vikrantgupta25 committed:
chore: move hostname to resource attributes for logs qf (#6303)

* chore: move hostname to resource attributes for logs qf

* chore: fix the key for hostname

YounixM committed:
feat: handle invite user flows

YounixM committed:
feat: handle onboarding visibility

YounixM committed:
chore: update org onboarding package

YounixM committed:
chore: remove workspace from fullscreen

YounixM committed:
feat: handle light mode and remove unnecessary logos

srikanthccv committed:
fix: add support for {{.Labels.<key>}} with dots in `key` for template (#6282)

dependabot[bot] committed:
chore(deps): bump uplot from 1.6.26 to 1.6.31 in /frontend

Bumps [uplot](https://github.com/leeoniya/uPlot) from 1.6.26 to 1.6.31.
- [Release notes](https://github.com/leeoniya/uPlot/releases)
- [Commits](https://github.com/leeoniya/uPlot/compare/1.6.26...1.6.31)

---
updated-dependencies:
- dependency-name: uplot
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

YounixM committed:
fix: handle redirect in onboarding (#6324)

shivanshuraj1333 committed:
fix: use query builder for metrics onboarding API (#6327)

YounixM committed:
fix: frontend/package.json & frontend/yarn.lock to reduce vulnerabilities (#6266)

The following vulnerabilities are fixed with an upgrade:
- https://snyk.io/vuln/SNYK-JS-HTTPPROXYMIDDLEWARE-8229906
- https://snyk.io/vuln/SNYK-JS-UPLOT-6209224

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

ahmadshaheer committed:
chore: upgrade axios from 1.7.4 to 1.7.7 (#6291)

ankitnayan committed:
[Snyk] Security upgrade alpine from 3.18.5 to 3.20.3 (#6237)

The following vulnerabilities are fixed with an upgrade:
- https://snyk.io/vuln/SNYK-ALPINE318-BUSYBOX-6913411
- https://snyk.io/vuln/SNYK-ALPINE318-BUSYBOX-7249265
- https://snyk.io/vuln/SNYK-ALPINE318-BUSYBOX-7249419
- https://snyk.io/vuln/SNYK-ALPINE318-OPENSSL-6152404
- https://snyk.io/vuln/SNYK-ALPINE318-OPENSSL-6152404

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

dependabot[bot] committed:
chore(deps): bump webpack from 5.88.2 to 5.94.0 in /frontend (#5813)

Bumps [webpack](https://github.com/webpack/webpack) from 5.88.2 to 5.94.0.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.88.2...v5.94.0)

---
updated-dependencies:
- dependency-name: webpack
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

shivanshuraj1333 committed:
fix: typo (#6334)

nityanandagohain committed:
fix: issues with resource query builder w.r.t quotes (#6318)

ahmadshaheer committed:
fix: redirect to docs on clicking alert setup guide in create alert page (#6265)

shivanshuraj1333 committed:
enable scenario 4 on staging (#6269)

* fix: enable env at docker compose

YounixM committed:
fix: add safety check to check if anomaly rule in uplot chart options (#6343)

YounixM committed:
feat: improve async handling for org onboarding cases (#6342)

srikanthccv committed:
chore: add feature flag, handle out-of-index error, some house keeping work (#6344)

YounixM committed:
fix: re add threshold for promql alerts (#6355)

YounixM committed:
chore: github wf update pr labels and block pr until related docs are shipped for the feature (#6333)

SagarRajput-7 committed:
feat: added generic UI for scenario 1,3,4 (#6287)

* feat: added generic table component for scenario 1,3,4

* feat: added generic logic for mq detail tables and consumed for sc-1,2

* feat: added overview and details table for scenario-3

* feat: added table row clicks func

* feat: resolved comments

SagarRajput-7 committed:
feat: added onboarding setup for Producer for Messaging queues (#6236)

* fix: added onboarding setup for producer/consumer for Messaging queues

* fix: polled onboarding status api

* feat: added onboarding status api with useQueury functions and updated endpoints

* feat: added onboarding status api util for attribute data

* feat: refactoring and url query changes

* feat: changed start and end time to nanosecond for api payload

* feat: added comment description

YounixM committed:
fix: initialize target to 3 in anomaly detection alert (#6362)

SagarRajput-7 committed:
feat: added healthcheck and attribute checklist component for Kafka (#6371)

* feat: added healthcheck and attribute checklist component for Kafka

* feat: corrected the onboardingapi payload

* feat: added missing configuration button at overview and onboarding flow

SagarRajput-7 committed:
feat: added kafka - scenario - 4 (#6370)

* feat: added kafka - scenario - 4

* feat: added error handling and correct api handler for kafka apis

SagarRajput-7 committed:
feat: added onboarding detail for consumer setup (#6372)

* feat: added onboarding detail for consumer setup

* feat: corrected spelling

* feat: reorders imports correctly

SagarRajput-7 committed:
feat: updated the design for Messaging Queue - summary section (#6319)

* feat: updated the design for Messaging Queue - summary section

* feat: resolved comments

* feat: added better logic for switch options and resolved query

nityanandagohain committed:
feat: support for scope in logs old and new qb (#6339)

SagarRajput-7 committed:
feat: corrected the handling of relativeTime as null in alertHistory page (#6392)

vikrantgupta25 committed:
feat: added support for instrumentation scope in logs (#6378)

* feat: added support for instrumentation scope in logs

* chore: remove console logs

* fix: the logic for rendering prefix

* feat: address review comments

---------

Co-authored-by: SagarRajput-7 <162284829+SagarRajput-7@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump express from 4.19.2 to 4.21.1 in /frontend (#6166)

Bumps [express](https://github.com/expressjs/express) from 4.19.2 to 4.21.1.
- [Release notes](https://github.com/expressjs/express/releases)
- [Changelog](https://github.com/expressjs/express/blob/4.21.1/History.md)
- [Commits](https://github.com/expressjs/express/compare/4.19.2...4.21.1)

---
updated-dependencies:
- dependency-name: express
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ahmadshaheer committed:
chore: setup router compatibility package (#6285)

ahmadshaheer committed:
fix: adding the key requires double enter before it gets added as label key after the first label (#6296)

SagarRajput-7 committed:
feat: added kafka - scenario - 4 - drop rate table (#6380)

* feat: added kafka - scenario - 4 - drop rate table

* feat: added api, new table and traceid redirection

* feat: code refactor

SagarRajput-7 committed:
fix: kafka - misc fix and features (#6379)

* feat: fixed multiple fixes and chores in kafka 2.0

* feat: fixed producer latency - producer-detail call

* feat: fixed mq-detail page layout and pagination

* feat: resolved comments

YounixM committed:
chore: update events for onboarding part 2 (#6397)

nityanandagohain committed:
Issue 6367 (#6376)

* fix: issue with orderby by materialized column

* fix: tests

* fix: order by issue in old explorer as well

prashant-shahi committed:
Merge branch 'main' into release/v0.58.x

prashant-shahi committed:
chore(signoz): pin versions: SigNoz 0.58.0, SigNoz OtelCollector 0.111.8

Signed-off-by: Prashant Shahi <prashant@signoz.io>

srikanthccv committed:
fix: test notification missing for anomaly alert (#6391)

prashant-shahi committed:
Merge pull request #6405 from SigNoz/release/v0.58.x

Release/v0.58.x

SagarRajput-7 committed:
feat: added metric page in messaging queues (#6399)

* feat: added metric page in messaging queues

* feat: added misc fixes

* feat: removed a class name from mqcards

* feat: added lightMode styles for kafka 2.0 (#6400)

* feat: resolved comments and used strings

YounixM committed:
[Snyk] Fix for 2 vulnerabilities (#6215)

* fix: frontend/package.json & frontend/yarn.lock to reduce vulnerabilities

The following vulnerabilities are fixed with an upgrade:
- https://snyk.io/vuln/SNYK-JS-UPLOT-6209224
- https://snyk.io/vuln/SNYK-JS-VUETEMPLATECOMPILER-8219888

* chore: upgrade design tokens to 1.1.3

---------

Co-authored-by: snyk-bot <snyk-bot@snyk.io>
Co-authored-by: ahmadshaheer <ashaheerki@gmail.com>

ahmadshaheer committed:
fix: fix the issue of adding new query in new alert page changing the data source (#6286)

Co-authored-by: Srikanth Chekuri <srikanth.chekuri92@gmail.com>

ahmadshaheer committed:
feat: alert rename interaction (#6208)

* feat: alert rename interaction

* feat: add support for enter and escape shortcuts to submit and cancel rename

* chore: add missing alert field

* chore: update the style similar to dashboard rename

* refactor: remove buttonProps

* chore: add missing alert property to fix the build

---------

Co-authored-by: Srikanth Chekuri <srikanth.chekuri92@gmail.com>

vikrantgupta25 committed:
feat: consume the new licenses v3 structure.  (#6341)

* feat: setup for licenses v3 integration

* feat: added some more logic

* feat: validator changes

* chore: added a couple of todos

* feat: added config parameter for licenses v3 and the boot option

* feat: some typo fix

* feat: added refresh licenses handler

* feat: handle the start manager license activation

* chore: text updates

* feat: added list licenses call

* chore: refactor the entire code to cleanup interfaces

* fix: nil pointer error

* chore: some minor edits

* feat: model changes

* feat: model changes

* fix: utilise factory pattern

* feat: added default basic plan

* chore: added test cases for new license function

* feat: added more test cases

* chore: make the licenses id not null

* feat: cosmetic changes

* feat: cosmetic changes

* feat: update zeus URL

* chore: license testing fixes

* feat: added license status and category handling for query-service

* chore: added v3 support in v2 endpoint

* chore: http response codes and some code cleanup

* chore: added detailed test cases

* chore: address review comments

* chore: some misc cleanup

Commit:
feat: fixProducerAPI (#6422)

chore: bugfix

nityanandagohain committed:
fix: add severity_text legend (#6415)

nityanandagohain committed:
fix: ignore ts for panel type table (#6419)

srikanthccv committed:
chore: add k8s {deployment, daemonset, statefulset, job} resources (#6401)

srikanthccv committed:
fix: update request payload for span metrics queries (#6323)

vikrantgupta25 committed:
chore: return proper http codes on unique constraint error (#6428)

srikanthccv committed:
chore: add experimental rate/increase calc (#6432)

prashant-shahi committed:
Merge branch 'develop' into release/v0.58.x

srikanthccv committed:
chore: add v2 metric writer to pipelines (#6345)

nityanandagohain committed:
feat: trace V4 QB (#6407)

* feat: trace V4 QB

* fix: update get column name and remove id

* fix: handle contains and update tests

* fix: remove unwanted step interval calculation

* fix: add test cases

* fix: add tests for static columns in QB

* fix: add more order by tests

* fix: update order by logic

YounixM committed:
feat: ingestion limits - add toggle feature (#6430)

YounixM committed:
fix: show org onboarding only to cloud customers (#6451)

SagarRajput-7 committed:
feat: removed dashboard uuid is all cases be it duplicate, empty or somevalid, while import json (#6448)

* feat: removed dashboard uuid is all cases be it duplicate, empty or somevalid, while import json

* feat: added comment to better explain the logic

prashant-shahi committed:
Merge branch 'develop' into release/v0.58.x

prashant-shahi committed:
chore(signoz): pin versions: SigNoz 0.58.1

Signed-off-by: Prashant Shahi <prashant@signoz.io>

prashant-shahi committed:
Merge branch 'main' into release/v0.58.x

nityanandagohain committed:
feat: support for window based pagination in new trace v4 (#6440)

* feat: support for window based pagination in new trace v4

* fix: update pagination logic

* fix: update comment

* fix: substract correct length

* fix: revert changes

* fix: add tests for querier

* fix: rename matcher

* fix: handle offset inmemory for list queries in traces

* fix: correct var name

* fix: add max pagination limit for traces

prashant-shahi committed:
Merge pull request #6452 from SigNoz/release/v0.58.x

Release/v0.58.1

nityanandagohain committed:
fix: update logs struct to fix live logs (#6453)

prashant-shahi committed:
Merge branch 'develop' into release/v0.58.x

nityanandagohain committed:
feat: support for new enrichment logic in traces (#6438)

* feat: support for new enrichment logic in traces

* fix: default test added

* fix: update func name in links

* Update pkg/query-service/utils/logs_test.go

Co-authored-by: ellipsis-dev[bot] <65095814+ellipsis-dev[bot]@users.noreply.github.com>

---------

Co-authored-by: ellipsis-dev[bot] <65095814+ellipsis-dev[bot]@users.noreply.github.com>

srikanthccv committed:
fix: compare op outside bounds for anomaly alert (#6458)

srikanthccv committed:
chore: add user email to log_comment (#6461)

srikanthccv committed:
chore: add missing alert telemetry (#6459)

YounixM committed:
[Snyk] Security upgrade alpine from 3.18.6 to 3.20.3 (#6463)

The following vulnerabilities are fixed with an upgrade:
- https://snyk.io/vuln/SNYK-ALPINE318-BUSYBOX-6913411
- https://snyk.io/vuln/SNYK-ALPINE318-BUSYBOX-7249236
- https://snyk.io/vuln/SNYK-ALPINE318-BUSYBOX-7249265
- https://snyk.io/vuln/SNYK-ALPINE318-BUSYBOX-7249265
- https://snyk.io/vuln/SNYK-ALPINE318-BUSYBOX-7249419

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

nityanandagohain committed:
fix: remove unwanted trace API's (#6464)

ahmadshaheer committed:
feat: add 'create channel' option in channels list and refetch alert channels on opening the channels dropdown (#6416)

* feat: add channel creation option and auto-refresh channels list on dropdown open

* chore: move inline styles to style.ts

* fix: show the prompt to ask admin if the user doesn't have permissions

* fix: display create channel option only if the user has permission

* fix: prevent repeated new alert event logs + log new channel option inside dropdown

vikrantgupta25 committed:
chore: use the license v2 key to fill licenses v3 on startup (#6468)

* feat: use the license v2 key to fill licenses v3 on startup

* chore: make the init only if the licenses v2 is present

* chore: address review comments

nityanandagohain committed:
fix: update TestListTsRange to return all range (#6470)

prashant-shahi committed:
Merge branch 'develop' into release/v0.58.x

srikanthccv committed:
fix: incorrect formula for apdex (#6460)

prashant-shahi committed:
Merge branch 'develop' into release/v0.58.x

prashant-shahi committed:
chore(signoz): pin versions: SigNoz 0.58.2

Signed-off-by: Prashant Shahi <prashant@signoz.io>

prashant-shahi committed:
Merge branch 'main' into release/v0.58.x

prashant-shahi committed:
Merge pull request #6471 from SigNoz/release/v0.58.x

Release/v0.58.2

vikrantgupta25 committed:
chore: set zeusURL via build time variable (#6475)

nityanandagohain committed:
Trace static fields and structs for trace v4. (#6469)

* fix: update static fields and add response structs

* fix: update ch names

* fix: move models to it's own file

therealpandey committed:
feat(key): enable edit access for gateway (#6480)

vikrantgupta25 committed:
chore: exactly match the list licenses v2 structure with FF on (#6481)

ahmadshaheer committed:
chore: add log events to alert history page (#6396)

* chore: pass active route with RouteTab's onChangeHandler

* chore: add log events to alert history page

---------

Co-authored-by: Srikanth Chekuri <srikanth.chekuri92@gmail.com>

YounixM committed:
chore: update http proxy middleware (#6486)

prashant-shahi committed:
Merge branch 'main' into release/v0.59.x

prashant-shahi committed:
chore(signoz): pin versions: SigNoz 0.59.0, SigNoz OtelCollector 0.111.9

Signed-off-by: Prashant Shahi <prashant@signoz.io>

prashant-shahi committed:
Merge pull request #6490 from SigNoz/release/v0.59.x

Release/v0.59.x

nityanandagohain committed:
fix: remove service overview API (#6495)

nityanandagohain committed:
feat: update clickhouse reader to support new trace schema (#6479)

* feat: update clickhouse reader to support new trace schema

* fix: minor fixes

* fix: address comments

* fix: add changes to overview function

* fix: add changes to overview function

* fix: use hardcoded true

* fix: address comments

prashant-shahi committed:
Merge branch 'main' into sync/post-release-v0.59

prashant-shahi committed:
Merge pull request #6498 from SigNoz/sync/post-release-v0.59

Sync/post release v0.59

YounixM committed:
feat: added the host list view and filters (#6210) (#6501)

* feat: added the host list view and filters (#6210)

* feat: added the host list view and filters

* feat: removed group by filter and added autocomplete for where clause

* feat: updated the table view and added the pagination

* feat: pass updated filters to api to get filtered data in the list

* feat: added global time range and order by for cpu,memory,iowait,load

* feat: added order by and color codes for cpu and memory usage progress bar

* refactor: removed inline styles

* Host lists improvement (#6366)

* style: added new style changes for date time selection in host lists view

* style: added padding to date time selector

* style: removed unnecessary styles for host tabs

* style: removed unused css

* feat: added the host detail view (#6267)

* Host containers (#6297)

* feat: added the host detail view

* feat: completed containers and processes details view

* Show host metrics panels in metrics tab. (#6306)

* feat: added the host detail view

* feat: completed containers and processes details view

* feat: added host metrics panels in metrics tabs

* refactor: removed inline styles from host containers and processes tabs

* style: added top and bottom margin to containers and processes tab

* Metrics time selection (#6360)

* feat: added the host detail view

* feat: completed containers and processes details view

* feat: added host metrics panels in metrics tabs

* refactor: removed inline styles from host containers and processes tabs

* feat: added logs and traces tab in host metrics detail view

* chore: removed console statements

* feat: added DateTimeSelection component in metrics tab

* style: added top and bottom margin to containers and processes tab

* style: removed inline styles

* feat: added logs and traces tab in host metrics detail view (#6359)

* feat: added the host detail view

* feat: completed containers and processes details view

* feat: added host metrics panels in metrics tabs

* refactor: removed inline styles from host containers and processes tabs

* feat: added logs and traces tab in host metrics detail view

* chore: removed console statements

* feat: added filters and time selection in traces tab

* fix: resolved metrics,logs and traces tab issues

* feat: added navigation for logs and traces to respective explorer pages

* fix: added the code for logs tab and navigation to respective explorer page

* fix: added fixes for date time selection custom issue

* style: added styles for light mode

* refactor: removed unused code and added comments

* refactor: added new code for host metric attribute keys

* feat: reset query data once we are on infra monitoring page

* chore: remove optional parameter from get attributes and groupby interfaces

* feat: update ui as per the designs

* fix: logs list, time select and other ui issues

* feat: update title for infra monitoring page

* feat: update copies

* feat: update styles for light mode

* fix: reset page size on filter, open explorers in new tab, enable horizontal scroll

* feat: traces tab updates

* feat: move infra monitoring behind ff

* fix: remove sorting from host listing page

---------

Co-authored-by: Yunus M <myounis.ar@live.com>

* chore: fix lint errors

---------

Co-authored-by: rahulkeswani101 <rahul@signoz.io>

nityanandagohain committed:
feat: ttl api for new trace tables (#6497)

* feat: tt api for new trace tables

* fix: remove print and use correct context

* fix: update var name

nityanandagohain committed:
feat: trace v4 integration (#6226)

* feat: trace v4 inital commit

* fix: add remaining files

* fix: integrate with querier

* fix: get trace by id api updated

* fix: add servicename resource filter

* fix: tests

* fix: use correct prepQUery

* fix: services page

* fix: minor fixes to use the new table in api's and querier

* fix: add support for window based pagination

* feat: support for faster trace detail

* fix: searchTraces

* fix: attribute enrichment updated and issue in group by

* fix: issues in group by

* fix: enrichment using alias

* fix: test file added

* fix: tests

* fix: group by with filters

* fix: add subquery

* fix: trigger builde

* fix: update pagination logic and few ch column names

* fix: update qb

* fix: add tests

* feat: minor fixes

* fix: update pagination logic

* fix: update pagination logic

* fix: remove utils

* fix: remove unwanted API's

* fix: attribute and attribute values v2

* fix: autocomplete api's updated

* fix: tests fixed

* feat: minor fixes

* fix: update telemetry functions

* fix: dont use alias, use proper col names

* fix: move models to it's own file

* fix: minor fixes

* fix: address comments

* fix: add to serviceoverview function

* fix: add changes to overview function

* fix: address comments

* fix: remove printlines

---------

Co-authored-by: Srikanth Chekuri <srikanth.chekuri92@gmail.com>

SagarRajput-7 committed:
fix: fixed back navigation issue for dashboard details coming from dashboard list table (#6496)

SagarRajput-7 committed:
fix: updated folder name and structure (#6488)

srikanthccv committed:
chore: enable sorting for hosts list (#6502)

srikanthccv committed:
chore: remove the fancy text while fetching data (#6505)

vikrantgupta25 committed:
fix: decrease the pad angle and remove the empty legend entries (#6507)

* fix: pie chart should not have padding between adjacent items

* fix: remove the labels which do not have any data

YounixM committed:
chore: update http proxy middleware (#6499)

nityanandagohain committed:
fix: add v2 for getServices and GetTopOperations (#6516)

* fix: add v2 for getServices and GetTopOperations

* fix: add comments

* fix: update logic for filters

---------

Co-authored-by: Srikanth Chekuri <srikanth.chekuri92@gmail.com>

nityanandagohain committed:
fix: add settings to allow distributed_product_mode for trace panel (#6517)

* fix: add settings to allow distributed_product_mode for trace panel

* fix: tests fixed

* fix: add 10gb memory limit

---------

Co-authored-by: Srikanth Chekuri <srikanth.chekuri92@gmail.com>

nityanandagohain committed:
fix: update api's to repect the new changes in attributes table (#6526)

srikanthccv committed:
chore: add additional info for host metrics onboarding (#6529)

vikrantgupta25 committed:
chore: move away from parallel usages of licenses v2 and v3 (#6527)

* chore: move all the usages of license v2 to license v3

* chore: added log lines for debugging

nityanandagohain committed:
fix: add telemetry for trace migration (#6537)

Ruppsn committed:
Fix for running signoz-schema-migrator under docker-swarm (#6489)

raj-k-singh committed:
Chore/qs use collector simulator from signoz otel collector (#6539)

* chore: qs: logs pipeline preview: use collectorsimulator from signoz-otel-collector

* chore: qs: remove collectorsimulator: located in signoz-otel-collector now

prashant-shahi committed:
chore(signoz): pin versions: SigNoz 0.60.0, SigNoz OtelCollector 0.111.13

Signed-off-by: Prashant Shahi <prashant@signoz.io>

prashant-shahi committed:
Merge pull request #6546 from SigNoz/release/v0.60.x

Release/v0.60.x

YounixM committed:
feat: improve empty hosts, incorrect metrics and no filter views (#6530)

* feat: improve empty hosts, incorrect metrics and no filter views

* feat: add infra monitoring - host lists - usage events (#6536)

* feat: add usasge events

* feat: add reqrest early access events

---------

Co-authored-by: Srikanth Chekuri <srikanth.chekuri92@gmail.com>

YounixM committed:
feat: handle keyboard navigations for column selection in logs explor… (#6548)

* feat: handle keyboard navigations for column selection in logs explorer options view

* chore: remove console log

vikrantgupta25 committed:
fix: explicit mouse move event user action (#6549)

* fix: explicit mouse move event user action

* fix: handle next element selection on click

* fix: handle the last 0 case

* fix: handle search reset

nityanandagohain committed:
fix: add dashbard names and update logs count in telemetry (#6553)

SagarRajput-7 committed:
fix: fixed typo in metric view dashboard title (#6557)

YounixM committed:
chore: update sentry react to v8 (#6569)

YounixM committed:
chore: remove infra monitoring ff (#6571)

vikrantgupta25 committed:
feat: show failed payment banner and block the workspace integration  (#6560)

* feat: added new API endpoint for fetching the active license

* feat: add setup for apis on frontend

* feat: frontend infrastructure changes for app context and workspace suspended

* feat: added workspace suspended component

* feat: send back to application if workspace is not suspended

* feat: added the missing creative

* chore: only move to suspended state when state is payment_failed

* chore: address review comments

* fix: tab naming

vikrantgupta25 committed:
fix: make the entire row clickable in logs quick filters (#6574)

prashant-shahi committed:
Merge branch 'main' into release/v0.61.x

prashant-shahi committed:
chore(signoz): pin versions: SigNoz 0.61.0, SigNoz OtelCollector 0.111.14

Signed-off-by: Prashant Shahi <prashant@signoz.io>

prashant-shahi committed:
Merge pull request #6591 from SigNoz/release/v0.61.x

Release/v0.61.x

amlannandy committed:
feat: add functionality to export dashboard as json from listing page

prashant-shahi committed:
Merge pull request #6593 from SigNoz/release-sync/v0.61.x

Release Sync/v0.61.x

nityanandagohain committed:
fix: enable smart trace detail (#6596)

amlannandy committed:
feat: address comments

amlannandy committed:
chore: replace the 'Get started' CTA with 'New source'

amlannandy committed:
chore: address comments

amlannandy committed:
feat: add unit test

amlannandy committed:
feat: update tests

amlannandy committed:
feat: update unit test

amlannandy committed:
feat: update unit test

amlannandy committed:
fix: resolve the unneccessary refetching of graphs on service details page

deepakdinesh1123 committed:
fix(docker): otel-collector-migrator command fixed in docker-compose-core.yaml (#6564)

Co-authored-by: Srikanth Chekuri <srikanth.chekuri92@gmail.com>

amlannandy committed:
chore: replace the 'Get started' CTA with 'New source'

shivanshuraj1333 committed:
feat: add byte rate to producer API (#6579)

* fix: fix partition check for topic throughput

amlannandy committed:
fix: resolve unnecessary refetching of graphs on service details page

nityanandagohain committed:
fix: smartTraceAlgo for new schema (#6602)

amlannandy committed:
chore: address comments

nityanandagohain committed:
fix: update default alert ch queries for traces and logs (#6552)

* fix: update default alert ch queries for traces and logs

* fix: use service.name instead of peer.service

* fix: add link to docs

* fix: add alias for service name

nityanandagohain committed:
fix: migration for ingestion dashboard (#6610)

* fix: migration for ingestion dashboard

* fix: remove redundant uuid

* Update pkg/query-service/app/dashboards/model.go

Co-authored-by: Srikanth Chekuri <srikanth.chekuri92@gmail.com>

---------

Co-authored-by: Srikanth Chekuri <srikanth.chekuri92@gmail.com>

nityanandagohain committed:
fix: fix count aggregate attribute column name (#6613)

* fix: fix count aggregate attribute column name

* fix: add test for column as well

* fix: use new schema in threshold rule enrichment

prashant-shahi committed:
Merge branch 'main' into release/v0.62.x

prashant-shahi committed:
chore(signoz): pin versions: SigNoz 0.62.0, SigNoz OtelCollector 0.111.15

Signed-off-by: Prashant Shahi <prashant@signoz.io>

prashant-shahi committed:
chore(signoz): pin versions: SigNoz 0.62.0, SigNoz OtelCollector 0.111.15

Signed-off-by: Prashant Shahi <prashant@signoz.io>

prashant-shahi committed:
Merge pull request #6625 from SigNoz/release/v0.62.x

Release/v0.62.x

SagarRajput-7 committed:
feat: added table column and row logic for the new api response structure for producer overview (#6433)

* feat: added table column and row logic for the new api response structure for prodcure overview

* feat: fixed typo in function name

* feat: consumed new 2 table merging logic for producer latency overview

* feat: added 3 digit precision to 'ingestion_rate' and 'byte_rate'in consumer overview

prashant-shahi committed:
chore: deprecate develop branch - use main

Signed-off-by: Prashant Shahi <prashant@signoz.io>

therealpandey committed:
feat(signoz-ingestion-key): rename signoz-access-token to signoz-ingestion-key (#6633)

dependabot[bot] committed:
chore(deps): bump golang.org/x/crypto from 0.27.0 to 0.31.0 (#6638)

Bumps [golang.org/x/crypto](https://github.com/golang/crypto) from 0.27.0 to 0.31.0.
- [Commits](https://github.com/golang/crypto/compare/v0.27.0...v0.31.0)

---
updated-dependencies:
- dependency-name: golang.org/x/crypto
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ahmadshaheer committed:
Feat: Timezone picker feature (#6474)

* feat: time picker hint and timezone picker UI with basic functionality + helper to get timezones

* feat: add support for esc keypress to close the timezone picker

* chore: add the selected timezone as url param and close timezone picker on select

* fix: overall improvement + add searchIndex to timezone

* feat: timezone preferences UI

* chore: improve timezone utils

* chore: change timezone item from div to button

* feat: display timezone in timepicker input

* chore: fix the typo

* fix: don't focus on time picker when timezone is clicked

* fix: fix the issue of timezone breaking for browser and utc timezones

* fix: display the timezone in timepicker hint 'You are at'

* feat: timezone basic functionality (#6492)

* chore: change div to fragment + change type to any as the ESLint complains otherwise

* chore: manage etc timezone filtering with an arg

* chore: update timezone wrapper class name

* fix: add timezone support to downloaded logs

* feat: add current timezone to dashboard list and configure metadata modal

* fix: add pencil icon next to timezone hint + change the copy to Current timezone

* fix: properly handle the escape button behavior for timezone picker

* chore: replace @vvo/tzdb with native Intl API for timezones

* feat: lightmode for timezone picker and timezone adaptation components

* fix: use normald tz in browser timezone

* fix: timezone picker lightmode fixes

* feat: display selected time range in 12 hour format

* chore: remove unnecessary optional chaining

* fix: fix the typo in css variable

* chore: add em dash and change icon for timezone hint in date/time picker

* chore: move pen line icon to the right of timezone offset

* fix: fix the failing tests

* feat: handle switching off the timezone adaptation

amlannandy committed:
Merge branch 'develop' into dashboard-list-export-json

amlannandy committed:
feat: add functionality to export dashboard as json from listing page

ahmadshaheer committed:
fix: improve traces table ux by making the table scrollable (#6423)

* fix: improve traces table ux by making the table scrollable

* chore: remove explicit scroll.x and modify the existing scroll.x from true to max-content

* fix(Traces Explorer): make trace explorer card full-width when collapsed, resizable when expanded

srikanthccv committed:
chore: add queries with tag attributes v1 (#6643)

srikanthccv committed:
chore: use tag attributes v2 table (#6616)

therealpandey committed:
feat(licenses): deprecate licenses v2 (#6626)

deprecate licenses v2

vikrantgupta25 committed:
fix: unable to remove query tags from the beginning when similar tags present  (#6645)

* fix: unable to remove query tags from the begining

* fix: focus only when the filters dropdown is opened

* fix: used auto focus prop

vikrantgupta25 committed:
fix: timerange selected for traces to logs (#6634)

vikrantgupta25 committed:
fix: correlation time stamps for APM to traces and logs (#6632)

therealpandey committed:
fix(flag): add missing flag back (#6647)

raj-k-singh committed:
Fix: QS: logs pipelines: better validation of pipelines being saved (#6652)

* chore: add test validating invalid field paths in pipeline operators are rejected

* chore: refactor posted pipelines validation to use a controller method

* fix: run a collector simulation to validate pipeline config being saved

* chore: minor cleanup

prashant-shahi committed:
Merge branch 'main' into release/v0.63.x

prashant-shahi committed:
chore(signoz): pin versions: SigNoz 0.63.0, SigNoz OtelCollector 0.111.16

Signed-off-by: Prashant Shahi <prashant@signoz.io>

prashant-shahi committed:
chore(signoz): pin versions: SigNoz 0.63.0, SigNoz OtelCollector 0.111.16

Signed-off-by: Prashant Shahi <prashant@signoz.io>

prashant-shahi committed:
Merge pull request #6662 from SigNoz/release/v0.63.x

Release/v0.63.x

nityanandagohain committed:
fix: add flags for using trace new schema (#6651)

shivanshuraj1333 committed:
Add support for trace_v3 schema in messaging queues (#6663)

feat: support trace v3 queries

prashant-shahi committed:
Merge branch 'main' into release/v0.64.x

prashant-shahi committed:
chore(signoz): pin versions: SigNoz 0.64.0

Signed-off-by: Prashant Shahi <prashant@signoz.io>

prashant-shahi committed:
chore(signoz): pin versions: SigNoz 0.64.0

Signed-off-by: Prashant Shahi <prashant@signoz.io>

prashant-shahi committed:
Merge pull request #6664 from SigNoz/release/v0.64.x

Release/v0.64.x

shivanshuraj1333 committed:
update service.instance.id (#6665)

* nit: update resource id and revert the flag

nityanandagohain committed:
fix: use placehold in limit and use proper exists (#6667)

shivanshuraj1333 committed:
fix: add bucketing (#6669)

Co-authored-by: Nityananda Gohain <nityanandagohain@gmail.com>

nityanandagohain committed:
feat: api for trace materialization (#6646)

* feat: api for trace materialization

* fix: minor changes and cleanup

* fix: minor fixes

* fix: update errors

* fix: address comments

* fix: address comments

SagarRajput-7 committed:
feat: changed start and end time logic for consumer lag details (#6605)

prashant-shahi committed:
Merge branch 'main' into chore/deprecate-develop

prashant-shahi committed:
Merge pull request #6668 from SigNoz/chore/deprecate-develop

chore: develop deprecation and related changes

srikanthccv committed:
chore: address some gaps in k8s monitoring (#6653)

srikanthccv committed:
chore: add pvcs list (#6654)

anabeto93 committed:
not sure how I messed this one up.

anabeto93 committed:
changes for the overrides due to the drastic changes made on develop.
